packages/editor/src/core/components/nice-dropdown.tsx

```diff
@@ -0,0 +1,123 @@
+import { EditorTooltip } from '@editor/editor-ui/editor-tooltip'
+import { faQuestionCircle } from '@fortawesome/free-regular-svg-icons'
+import { faCaretDown, faCaretUp } from '@fortawesome/free-solid-svg-icons'
+import * as Select from '@radix-ui/react-select'
+import { useEffect, useRef, useState } from 'react'
+
+import { FaIcon } from '@/components/fa-icon'
+import { cn } from '@/helper/cn'
+
+interface NiceDropdownProps {
+  label: string
+  helpText: string
+  onChange?: (val: string) => void
+  options: { label: string; value: string | number }[]
+  defaultValue?: string
+  className?: string
+  isConfirmed?: boolean
+}
+
+export const NiceDropdown = ({
+  label,
+  helpText,
+  onChange,
+  defaultValue,
+  options,
+  className,
+  isConfirmed: initialConfirmed,
+}: NiceDropdownProps) => {
+  const [selectedValue, setSelectedValue] = useState(defaultValue)
+  const [isOpen, setIsOpen] = useState(false)
+  const [triggerWidth, setTriggerWidth] = useState(0)
+  const [isConfirmed, setIsConfirmed] = useState(initialConfirmed ?? false)
+  const [hasOpened, setHasOpened] = useState(false)
+
+  const triggerRef = useRef<HTMLButtonElement>(null)
+
+  const handleChange = (newValue: string) => {
+    setSelectedValue(newValue)
+    onChange?.(newValue)
+    setIsConfirmed(true)
+  }
+
+  useEffect(() => {
+    if (triggerRef.current) {
+      setTriggerWidth(triggerRef.current.offsetWidth)
+    }
+  }, [])
+
+  useEffect(() => {
+    if (isOpen && !hasOpened) {
+      setHasOpened(true)
+    } else if (!isOpen && hasOpened) {
+      setIsConfirmed(true)
+    }
+  }, [isOpen, hasOpened])
+
+  // Update selectedValue if defaultValue changes
+  useEffect(() => {
+    setSelectedValue(defaultValue)
+  }, [defaultValue])
+
+  return (
+    <label
+      className={`h-100 mx-auto mb-0 mt-5 flex items-center text-almost-black ${className}`}
```

Comment:
works but that's also a good place to use the `cn` helper

Reply:
Done